### DISCUSSION TITLE
Does elastiknn supports index patterns ?

### DISCUSSION THREAD

**Discussion by flo-dhalluin** [alexklibisz/elastiknn#702] [Q&A]
Hi,

I would like to use the "index patterns" feature of ES, say creating a daily index named "vectors-YYYY-MM-DD" with and associated index template mapping : "my-vectors-mapping" 
{
   "patterns": ["vectors-*"],
   "mappings": {
           ...  
           "v": {"type": "elastiknn_dense_float_vector", "elastiknn":{..}}
    }
}

From my first tests, it seems that the index template is not correctly applied to created indexes, thus knn search on  /vectors_*/search fails.

Such a pattern would allow me to quickly expire data from the db by deleting "old" indexes. 

Thanks !

**Comment by alexklibisz** [comment]
Hi, I've never used index patterns. I also don't remember seeing anything related to index patterns while implementing the plugin.

How exactly is it failing?

**Comment by flo-dhalluin** [comment]
It seems that the mapping is not applied to dynamically created indexes ( I see all the mappings defined in the template applied to the fields, except the one for the elastiknn vectors ... ).  
And a nearest neighors query on /vectors_*/search leads to a crash ...

**Comment by alexklibisz** [comment]
Got it. Can you please post the error you're getting when you search? I'll open an issue.
